.github/actions/setup-deps/action.yml

```
@@ -0,0 +1,64 @@
+name: "Turborepo CI Setup"
+description: "Sets up everything for CI"
+inputs:
+  github-token:
+    description: "Github token. You can pass secrets.GITHUB_TOKEN"
+    required: true
+  setup-node:
+    description: "Set to false to prevent Node setup steps"
+    required: false
+    default: true
+  go-version:
+    description: "Change Go version"
+    required: false
+    default: "1.18"
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ inputs.go-version }}
+        cache: true
+        cache-dependency-path: cli/go.sum
+
+    - name: Set Up Protoc
+      uses: arduino/setup-protoc@v1
+      with:
+        version: "3.x"
+        repo-token: ${{ inputs.github-token }}
+
+    - name: Set Up GRPC protobuf
+      shell: bash
+      run: |
+        go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28.0
+        go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2.0
+
+    # Setup Node things
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2.2.2
+      if: ${{ inputs.setup-node == 'true' }}
```

Ordering suggestion (if it works).
```suggestion
      if: ${{ inputs.setup-node == 'true' }}
      uses: pnpm/action-setup@v2.2.2
```